### PR TITLE
expression: implement vectorized evaluation for `builtinIntIsNullSig`

### PR DESCRIPTION
**Pull request by tangwz**
### What problem does this PR solve? 
Implement vectorized evaluation for builtinIntIsNullSig.
Issue: #12103

### What is changed and how it works?
```
BenchmarkVectorizedBuiltinOpFunc/builtinIntIsNullSig-VecBuiltinFunc-4             540609              2086 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinOpFunc/builtinIntIsNullSig-NonVecBuiltinFunc-4           60596             19653 ns/op               0 B/op          0 allocs/op

```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_op_vec.go [hunk 1]
```diff
@@ -119,11 +119,31 @@ func (b *builtinUnaryMinusDecimalSig) vecEvalDecimal(input *chunk.Chunk, result
...
 
 func (b *builtinIntIsNullSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinIntIsNullSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	numRows := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETInt, numRows)
```
Comment (qw4990) [inline expression/builtin_op_vec.go]: How about reusing the `result` instead of allocating a new buffer.

**Comment by tangwz** [comment]
@qw4990 PTAL

**Comment by qw4990** [comment]
@tangwz Please resolve conflicts at your convenience.

**Pull request merged**